pkg/credentials/stsflow/stsflow.go

```diff
@@ -0,0 +1,263 @@
+
+	// WebIdentityTokenPath mount present on operator CSV
+	WebIdentityTokenPath = "/var/run/secrets/openshift/serviceaccount/token"
+
+	// Cloud Provider Vars
+	// https://github.com/openshift/cloud-credential-operator/blob/6a880b473554aee4d1f3cd125048fef2bca6a04d/pkg/azure/const.go#L4
+	AzureClientID           = "azure_client_id"
+	AzureClientSecret       = "azure_client_secret"
+	AzureRegion             = "azure_region"
+	AzureResourceGroup      = "azure_resourcegroup"
+	AzureResourcePrefix     = "azure_resource_prefix"
+	AzureSubscriptionID     = "azure_subscription_id"
+	AzureTenantID           = "azure_tenant_id"
+	AzureFederatedTokenFile = "azure_federated_token_file"
+
+	// https://github.com/openshift/cloud-credential-operator/blob/6a880b473554aee4d1f3cd125048fef2bca6a04d/pkg/gcp/actuator/actuator.go#L57
+	GcpSecretJSONKey = "service_account.json"
+
+	VeleroAWSSecretName   = "cloud-credentials"
+	VeleroAzureSecretName = "cloud-credentials-azure"
+	VeleroGCPSecretName   = "cloud-credentials-gcp"
+)
+
+// STSStandardizedFlow creates secret for Short Term Service Account Tokens from the environment variables for
+// AWS STS, GCP WIF, and Azure
+// User are to provides them during web console installation, env var will be set to operator deployment
+// during installation via OLM.
+// Returns "", errors if secret creation fails.
+// Returns <secretName>, nil if secret creation succeed
+// Returns "", nil there is no STS env provided.
+func STSStandardizedFlow() (string, error) {
+	// Reference of provided envs from web console.
+	// https://github.com/openshift/console/blob/f11a6158ae722200d342519971af337f8ff61d3a/frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-subscribe.tsx#L502-L555
+
+	// AWS STS environment variables
+	roleARN := os.Getenv(RoleARNEnvKey)
+
+	// GCP WIF environment variables
+	serviceAccountEmail := os.Getenv(ServiceAccountEmailEnvKey)
+	projectNumber := os.Getenv(ProjectNumberEnvKey)
+	poolId := os.Getenv(PoolIDEnvKey)
+	providerId := os.Getenv(ProviderId)
+
+	// Azure environment variables
+	clientID := os.Getenv(ClientIDEnvKey)
+	tenantID := os.Getenv(TenantIDEnvKey)
+	subscriptionID := os.Getenv(SubscriptionIDEnvKey)
+
+	// Check if any cloud provider credentials are provided
+	hasAWSCreds := len(roleARN) > 0
+	hasGCPCreds := len(serviceAccountEmail) > 0 &&
+		len(projectNumber) > 0 && len(poolId) > 0
+	hasAzureCreds := len(clientID) > 0 && len(tenantID) > 0 && len(subscriptionID) > 0
+
+	// If no credentials are provided, return nil
+	if !hasAWSCreds && !hasGCPCreds && !hasAzureCreds {
+		return "", nil
+	}
+	// Logger set from cmd/main.go by ctrl.SetLogger
+	logger := log.Log
+	installNS := os.Getenv("WATCH_NAMESPACE")
+	kcfg := pkgclient.GetKubeconf()
+	if hasAWSCreds {
+		CreateOrUpdateSTSAWSSecret(logger, roleARN, installNS, kcfg)
+	} else if hasGCPCreds {
+		CreateOrUpdateSTSGCPSecret(logger, serviceAccountEmail, projectNumber, poolId, providerId, installNS, kcfg)
+	}
+
```

Comment:
add azure in future pr and test there.. or could add now without testing so gcp can merge first..